packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -913,15 +918,12 @@ class AsyncTimelineEvent extends TimelineEvent {
   bool get hasOverlappingChildren {
     if (_hasOverlappingChildren != null) return _hasOverlappingChildren;
     for (int i = 0; i < children.length; i++) {
-      final currentChild = children[i];
+      final currentChild = children[i] as AsyncTimelineEvent;
       // We do not have to look back because children will be ordered by their
       // start times.
       for (int j = i + 1; j < children.length; j++) {
-        final sibling = children[j];
-        // TODO(kenz): Check for deep timestamp overlap - since these events are
-        // async, children can extend the time bound of their parents, meaning
-        // we may still have display collisions even if parents do not overlap.
-        if (currentChild.time.overlaps(sibling.time)) {
+        final sibling = children[j] as AsyncTimelineEvent;
```

Comment:
avoid `as` unless it is the only valid way to do it. Dart2js will be better at compiling out the type check if you don't use as when we ship an optimized build.